src/dissect/dissect.c

```diff
@@ -127,6 +132,14 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_HASH_DEVICE: {
+                        arg_hash_device = strdup(optarg);
```

Comment:
leaks memory if arg_hash_device was already set. Please use: 

```c
if (free_and_strdup(&arg_hash_device, optarg) < 0)
        return log_oom();
```

Reply:
Ah I didn't think about the case where it's passed multiple times - fixed, thanks